tests/testGradingMethods.js

```diff
@@ -0,0 +1,459 @@
+ */
+const loadHomeworkPage = async (user) => {
+  setUser(user);
+  const studentCourseInstanceUrl = baseUrl + '/course_instance/1';
+  let hm9InternalExternalManaulUrl = null;
+  const courseInstanceBody = await (await fetch(studentCourseInstanceUrl)).text();
+  const $courseInstancePage = cheerio.load(courseInstanceBody);
+  hm9InternalExternalManaulUrl =
+    siteUrl +
+    $courseInstancePage(
+      'a:contains("Homework for Internal, External, Manual grading methods")'
+    ).attr('href');
+  let res = await fetch(hm9InternalExternalManaulUrl);
+  assert.equal(res.ok, true);
+  return res.text();
+};
+
+describe('Grading method(s)', function () {
+  this.timeout(80000);
+
+  let $hm1Body = null;
+  let iqUrl = null;
+  let gradeRes = null;
+  let iqId = null;
+  let questionsPage = null;
+  let $questionsPage = null;
+
+  before('set up testing server', helperServer.before());
+  after('shut down testing server', helperServer.after);
+
+  after('reset default user', () => setUser(defaultUser));
+
+  describe('`gradingMethod` configuration (deprecated, backwards compatible)', () => {
+    describe('"Internal"', () => {
+      describe('"grade" action', () => {
+        before(
+          'load page as student and submit "grade" action to "Internal" type question',
+          async () => {
+            const hm1Body = await loadHomeworkPage(mockStudents[0]);
+            $hm1Body = cheerio.load(hm1Body);
+            iqUrl =
+              siteUrl +
+              $hm1Body('a:contains("HW9.1. Internal Grading: Adding two numbers")').attr('href');
+
+            // open page to produce variant because we want to get the correct answer
+            await fetch(iqUrl);
+            // get variant params
+            iqId = parseInstanceQuestionId(iqUrl);
+            const variant = (await sqlDb.queryOneRowAsync(sql.get_variant_by_iq, { iqId })).rows[0];
+
+            gradeRes = await saveOrGrade(
+              iqUrl,
+              { c: variant.params.a + variant.params.b },
+              'grade'
+            );
+            assert.equal(gradeRes.status, 200);
+
+            questionsPage = await gradeRes.text();
+            $questionsPage = cheerio.load(questionsPage);
+          }
+        );
+        it('should result in 1 grading jobs', async () => {
+          const grading_jobs = (await sqlDb.queryAsync(sql.get_grading_jobs_by_iq, { iqId })).rows;
+          assert.lengthOf(grading_jobs, 1);
+        });
+        it('should result in 1 "pastsubmission-block" component being rendered', () => {
+          assert.lengthOf($questionsPage('.pastsubmission-block'), 1);
+        });
+        it('should be given submission grade in "pastsubmission-block"', async () => {
+          assert.include(
+            questionsPage,
+            'Submitted answer\n          \n        </span>\n        <span>\n    \n        <span class="badge badge-success">correct: 100%</span>'
```

Comment:
Note to self: update all of these tests to not use this huge fragile string. Instead, we should probably use Cheerio to pull the information we're looking for out of the document.

Reply:
Done in eb62b1d.